cli/src/semgrep/core_runner.py

```diff
@@ -651,27 +653,29 @@ def plan_core_run(
             for language in rule.languages:
                 targets = list(
                     target_manager.get_files_for_rule(
-                        language,
-                        rule.includes,
-                        rule.excludes,
-                        rule.id,
+                        language, rule.includes, rule.excludes, rule.id, rule.product
                     )
                 )
 
                 for target in targets:
                     if all_targets is not None:
                         all_targets.add(target)
-                    target_info[target, language].append(rule_num)
+                    rules_nums, products = target_info[target, language]
+                    rules_nums.append(rule_num)
+                    products.add(rule.product.to_json_string())
 
         return Plan(
             [
                 Task(
                     path=target,
                     analyzer=language,
+                    products=tuple(
+                        map(lambda x: out.Product.from_json_string(x), products)
+                    ),
```

Comment:
Is it better to do this?
```suggestion
                    tuple(out.Product.from_json_string(x) for x in products)
```